lib/index.js

```diff
@@ -393,7 +401,7 @@ Mongoose.prototype.model = function(name, schema, collection, skipInit) {
   }
 
   var connection = options.connection || this.connection;
-  model = this.Model.compile(name, schema, collection, connection, this);
+  model = this.Model.compile(model || name, schema, collection, connection, this);
```

Comment:
You're right. I just saw the diff and thought it wasn't necessary based on my misunderstanding of what the `compile` method was doing.